### PR TITLE
executor: fix a problem that may cause system hang.

### PR DESCRIPTION
**Pull request by hanfei1991**
When the "fetch big data" check the finish is false and break, it will enter an infinite loop. it should return directly.
@coocood @shenli @zimulala @XuHuaiyu @winoros PTAL

### CODE REVIEW

File: executor/executor.go [hunk 1]
```diff
@@ -562,7 +562,7 @@ func (e *HashJoinExec) fetchBigExec() {
 		done := false
 		for i := 0; i < curBatchSize; i++ {
 			if e.finished.Load().(bool) {
-				break
+				return
```
Comment (shenli) [inline executor/executor.go]: Should we put the rows into channel? See line 580.

Comment (hanfei1991) [inline executor/executor.go]: No, we shouldn't. At that time, all join workers have stopped.

**Comment by shenli** [comment]
LGTM

**Comment by coocood** [comment]
LGTM

**Pull request merged**